tests/aws_iot_devicedefender_client_test.c

```diff
@@ -232,6 +280,20 @@ int main(int argc, char **argv) {
             AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("RaspberryPi"), /* TODO: make cli arg so policies can work */
         .task_period_ns = 5ul * 60ul * 1000000000ul};
     args.task_config = task_config;
+    aws_array_list_init_dynamic(
+        &args.task_config.custom_metrics, allocator, 0, sizeof(struct defender_custom_metric *));
+
+    ASSERT_SUCCESS(aws_iotdevice_defender_register_number_metric(
+        &args.task_config, allocator, "TestCustomMetricNumber", get_number_metric, allocator));
+
+    ASSERT_SUCCESS(aws_iotdevice_defender_register_number_list_metric(
+        &args.task_config, allocator, "TestCustomMetricNumberList", get_number_list_metric, allocator));
+
+    ASSERT_SUCCESS(aws_iotdevice_defender_register_string_list_metric(
+        &args.task_config, allocator, "TestCustomMetricStringList", get_string_list_metric, allocator));
+
+    ASSERT_SUCCESS(aws_iotdevice_defender_register_ip_list_metric(
```

Comment:
All examples are IPv4, is that expected?

Reply:
No. Will change one to IPv6 They are just strings so CRT isn't going to do any validation on the formatting of these strings.